### ISSUE TITLE
Flow annotations and no-undef + no-unused-vars

### ISSUE THREAD

**Issue by pnevyk** [babel/babel-eslint#625]
Hi,

I'd like to ask what is the status of handling `no-undef` and `no-unused-vars` of Flow type annotations directly in babel-eslint. If I understand it correctly, you started to use new eslint's scope API in [lib/analyze-scope.js](https://github.com/babel/babel-eslint/blob/master/lib/analyze-scope.js) to implement a custom Referencer which should address these issues (or at least `no-undef`).

With babel-eslint@8.2.3, I identified these snippets as violating `no-undef` rule:

```
opaque type Foo = number;
type Foo = { foo<T>(x: T): T };
declare type Foo = number;
declare interface IFoo {}
```

And these are causing `no-unused-vars` error:

```
declare class Foo {}
declare module Foo {}
declare var Foo: number;
declare function foo(): void;
```

If babel-eslint implements these, corresponding rules in eslint-plugin-flowtype can be deprecated and  removed eventually, as mentioned in [this](https://github.com/gajus/eslint-plugin-flowtype/issues/336) issue.

Is this a possibility or I am missing something?

**Comment by pascalduez** [comment]
I'm having a similar issue:

```js
// @flow

import type { Foo } from './foo';

type Whatever<F = Foo> = {
  foo: F,
};
```

```
error  'Foo' is defined but never used  no-unused-vars
```

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
